search/collector/eligible.go

```diff
@@ -23,28 +23,40 @@ import (
 	index "github.com/blevesearch/bleve_index_api"
 )
 
+type eligibleStore struct {
+	ids []index.IndexInternalID
+}
+
+func (s *eligibleStore) AddDocumentMatch(doc *search.DocumentMatch) *search.DocumentMatch {
+	copyOfID := make([]byte, len(doc.IndexInternalID))
+	copy(copyOfID, doc.IndexInternalID)
+	s.ids = append(s.ids, copyOfID)
```

Comment:
Is the copy above necessary? Wouldn't append allocate and copy implicitly?

Reply:
slices are reference types. This means they need to be copied explicitly to avoid changing the underlying slice.